### PR TITLE
update the translation of `Array.findLast()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

翻译`Array.findlast`

### Motivation

同上

### Additional details

无

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 1]
```diff
@@ -5,16 +5,16 @@ slug: Web/JavaScript/Reference/Global_Objects/Array/findLast
...
 
-如果你需要查找的方法：
+如果你需要找到：
 
 - *第一个*匹配的元素，使用 {{jsxref("Array.findIndex", "find()")}}。
 - 数组中最后一个匹配元素的*索引*，使用 {{jsxref("Array.findLastIndex", "findLastIndex()")}}。
-- *值的索引*，使用 {{jsxref("Array.prototype.indexOf()")}}。（它类似于 {{jsxref("Array.findIndex", "findIndex()")}}，但是会检查每个元素是否和值相等，而不是使用一个测试函数。）
-- 一个值是否*包含*在该数组中，使用 {{jsxref("Array.prototype.includes()")}}。同样地，它检查每个元素是否和值相等，而不是使用一个测试函数。
+- *值的索引*，使用 {{jsxref("Array.prototype.indexOf()")}}。（它类似于 {{jsxref("Array.findIndex", "findIndex()")}}，但是会检查每个元素是否与值相等，而不是使用一个测试函数。）
+- 该数组中是否*存在* 一个值，使用 {{jsxref("Array.prototype.includes()")}}。同样地，它检查每个元素是否和值相等，而不是使用一个测试函数。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:17]:
```suggestion
- 该数组中是否*存在*一个值，使用 {{jsxref("Array.prototype.includes()")}}。同样地，它检查每个元素是否和值相等，而不是使用一个测试函数。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 2]
```diff
@@ -27,38 +27,37 @@ findLast(callbackFn, thisArg)
...
+  - : 数组中测试元素的函数。回调必须返回一个[真值](/zh-CN/docs/Glossary/Truthy)，表示已找到匹配的元素，否则返回一个[假值](/zh-CN/docs/Glossary/Falsy)。函数在被调用时会传递以下参数：
     - `element`
       - : 当前遍历到的元素。
     - `index`
       - : 当前遍历到的元素的索引（位置）。
     - `array`
       - : 调用 `findLast()` 的数组本身。
 - `thisArg` {{optional_inline}}
-  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的对象。
+  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的值。详见[迭代方法](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods)
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:38]:
```suggestion
  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的值。详见[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods)
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 3]
```diff
@@ -27,38 +27,37 @@ findLast(callbackFn, thisArg)
...
 
 ### 返回值
 
 数组中满足提供的测试函数索引最高的元素；如果没有元素匹配，返回 {{jsxref("undefined")}}。
 
 ## 描述
 
-`findLast()` 方法对数组每一个元素按降序（索引从大到小）执行 `callbackFn` 函数，直到 `callbackFn` 返回一个[真值](/zh-CN/docs/Glossary/Truthy)。然后 `findLast()` 返回该元素的值并停止遍历数组。如果 `callbackFn` 没有返回一个真值，则 `findLast()` 返回 {{jsxref("undefined")}}。
+`findLast()` 是一个[迭代方法](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods)。该方法对数组每一个元素按降序（索引从大到小）执行 `callbackFn` 函数，直到 `callbackFn` 返回一个[真值](/zh-CN/docs/Glossary/Truthy)。然后 `findLast()` 返回该元素的值并停止遍历数组。如果 `callbackFn` 没有返回一个真值，则 `findLast()` 返回 {{jsxref("undefined")}}。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:46]:
```suggestion
`findLast()` 是一个[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#迭代方法)。该方法对数组每一个元素按降序（索引从大到小）执行 `callbackFn` 函数，直到 `callbackFn` 返回一个[真值](/zh-CN/docs/Glossary/Truthy)。然后 `findLast()` 返回该元素的值并停止遍历数组。如果 `callbackFn` 没有返回一个真值，则 `findLast()` 返回 {{jsxref("undefined")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 4]
```diff
@@ -27,38 +27,37 @@ findLast(callbackFn, thisArg)
...
 
-`callbackFn` 会为数组中的每个元素调用，而不仅仅是那些被赋值的元素，这意味着对于稀疏数组来说，该方法的效率要低于那些只遍历有值的索引的方法。
+`callbackFn` 会被数组中的*每个*元素调用，而不仅仅是那些被赋值的元素。对于[稀疏数组]()来说，空槽行为和`undefined`相同。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:48]:
```suggestion
`callbackFn` 会被数组中的*每个*元素调用，而不仅仅是那些被赋值的元素。对于[稀疏数组](/zh-CN/docs/Web/JavaScript/Guide/Indexed_collections#稀疏数组)来说，空槽行为和`undefined`相同。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 5]
```diff
@@ -27,38 +27,37 @@ findLast(callbackFn, thisArg)
...
 
-如果为 `findLast()` 提供了 `thisArg` 参数，它将在每次调用 `callbackFn` 时作为 `this` 值。如果没有被提供，则使用 {{jsxref("undefined")}}。
-
-`findLast()` 方法不会改变调用它的数组，但是提供的 `callbackFn` 可以。`findLast()` 处理的元素是在第一次调用 `callbackFn` *之前*设置的。因此：
+`findLast()` 方法不会改变调用它的数组，但是提供的 `callbackFn` 可以。但是请注意，数组的长度是在第一次调用`callbackFn`*之前*保存的。因此：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:50]:
```suggestion
`findLast()` 方法不会改变调用它的数组，但是提供的 `callbackFn` 可以。但是请注意，数组的长度是在第一次调用 `callbackFn` *之前*保存的。因此：
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 6]
```diff
@@ -27,38 +27,37 @@ findLast(callbackFn, thisArg)
...
 
 - `callbackFn` 不会访问在调用 `findLast()` 开始后才添加到数组中的任何元素。
 - 给已访问过的索引重新赋值将不会被 `callbackFn` 重新访问。
 - 给初始的范围外的索引赋值，其将不会被 `callbackFn` 访问。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:54]:
```suggestion
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 7]
```diff
@@ -27,38 +27,37 @@ findLast(callbackFn, thisArg)
...
-- 如果 `callbackFn` 更改了数组中现有的、尚未访问的元素，则其传递给 `callbackFn` 的值将是 `findLast()` 访问该元素索引时的值。
-- 仍然会访问已{{jsxref("Operators/delete", "删除")}}的元素。
+- 如果 `callbackFn` 更改了数组中现有的、尚未访问的元素，则其传递给 `callbackFn` 的值将是 `findLast()` 访问该元素索引时的值。[已删除](/en-US/docs/Web/JavaScript/Reference/Operators/delete)元素会被当做`undefined`来访问。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:55]:
```suggestion
- 如果 `callbackFn` 更改了数组中现有的、尚未访问的元素，则其传递给 `callbackFn` 的值将是 `findLast()` 访问该元素索引时的值。[已删除](/zh-CN/docs/Web/JavaScript/Reference/Operators/delete)的元素会被当做 `undefined` 来访问。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 8]
```diff
@@ -27,38 +27,37 @@ findLast(callbackFn, thisArg)
...
 
 >**警告：** 上一段描述的并发修改的情况经常导致难以理解的代码，通常应该避免（特殊情况除外）。
 
+`findLast()`方法是[通用的](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#generic_array_methods)。它只期望`this`值具有`length`属性和整数键的属性
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:59]:
```suggestion
`findLast()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它只期望 `this` 值具有 `length` 属性和整数键的属性
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 9]
```diff
@@ -120,22 +119,29 @@ function isPrime(element) {
 console.log([4, 6, 8, 12].findLast(isPrime)); // undefined, not found
 console.log([4, 5, 7, 8, 9, 11, 12].findLast(isPrime)); // 11
 ```
+### 在稀疏数组上使用findLast()
 
-### 访问不存在和删除的元素
+稀疏数组中的空槽*被*访问，并被视为`undefined`
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:124]:
```suggestion
稀疏数组中的空槽*被*访问，并被视为 `undefined`
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 10]
```diff
@@ -144,22 +150,30 @@ array.findLast(function(value, index) {
...
+// Visited index 6 with value 6
+// Visited index 5 with value undefined
+// Visited index 4 with value undefined
+// Visited index 3 with value undefined
+// Visited index 2 with value undefined
+// Visited index 1 with value 1
+// Visited index 0 with value 0
+```
+
+### 在非数组对象上调用findLast()
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:163]:
```suggestion
### 在非数组对象上调用 findLast()
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 11]
```diff
@@ -144,22 +150,30 @@ array.findLast(function(value, index) {
...
+
+`findLast()`方法读取`this`的`length`属性，然后访问每个整数索引
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:165]:
```suggestion
`findLast()` 方法读取 `this` 的 `length` 属性，然后访问每个整数索引
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 12]
```diff
@@ -173,8 +187,13 @@ array.findLast(function(value, index) {
 ## 参见
 
 - [Polyfill of `Array.prototype.findLast` in `core-js`](https://github.com/zloirock/core-js#ecmascript-array)
-- {{jsxref("Array.prototype.findLastIndex()")}} – find last element and return its index
-- {{jsxref("Array.prototype.includes()")}} – test whether a value exists in the array
-- {{jsxref("Array.prototype.filter()")}} – remove all non-matching elements
-- {{jsxref("Array.prototype.every()")}} – test all elements
-- {{jsxref("Array.prototype.some()")}} – test until one element matches
+- [Indexed collections](/en-US/docs/Web/JavaScript/Guide/Indexed_collections)
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:190]:
```suggestion
- [索引集合类](/zh-CN/docs/Web/JavaScript/Guide/Indexed_collections)
```

**Review by JasonLamv-t — CHANGES_REQUESTED** [review]
Some suggestions

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 13]
```diff
@@ -27,38 +27,36 @@ findLast(callbackFn, thisArg)
...
+  - : 数组中测试元素的函数。回调必须返回一个[真值](/zh-CN/docs/Glossary/Truthy)，表示已找到匹配的元素，否则返回一个[假值](/zh-CN/docs/Glossary/Falsy)。函数在被调用时会传递以下参数：
     - `element`
       - : 当前遍历到的元素。
     - `index`
       - : 当前遍历到的元素的索引（位置）。
     - `array`
       - : 调用 `findLast()` 的数组本身。
 - `thisArg` {{optional_inline}}
-  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的对象。
+  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的值。详见[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods)
```
Comment (fwqaaq) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:38]:
```suggestion
  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的值。详见[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods)。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 14]
```diff
@@ -27,38 +27,36 @@ findLast(callbackFn, thisArg)
...
 
 数组中满足提供的测试函数索引最高的元素；如果没有元素匹配，返回 {{jsxref("undefined")}}。
 
 ## 描述
 
-`findLast()` 方法对数组每一个元素按降序（索引从大到小）执行 `callbackFn` 函数，直到 `callbackFn` 返回一个[真值](/zh-CN/docs/Glossary/Truthy)。然后 `findLast()` 返回该元素的值并停止遍历数组。如果 `callbackFn` 没有返回一个真值，则 `findLast()` 返回 {{jsxref("undefined")}}。
+`findLast()` 是一个[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#迭代方法)。该方法对数组每一个元素按降序（索引从大到小）执行 `callbackFn` 函数，直到 `callbackFn` 返回一个[真值](/zh-CN/docs/Glossary/Truthy)。然后 `findLast()` 返回该元素的值并停止遍历数组。如果 `callbackFn` 没有返回一个真值，则 `findLast()` 返回 {{jsxref("undefined")}}。
 
-`callbackFn` 会为数组中的每个元素调用，而不仅仅是那些被赋值的元素，这意味着对于稀疏数组来说，该方法的效率要低于那些只遍历有值的索引的方法。
+`callbackFn` 会被数组中的*每个*元素调用，而不仅仅是那些被赋值的元素。对于[稀疏数组](/zh-CN/docs/Web/JavaScript/Guide/Indexed_collections#稀疏数组)来说，空槽行为和`undefined`相同。
```
Comment (fwqaaq) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:48]:
```suggestion
`callbackFn` 会被数组中的*每个*元素调用，而不仅仅是那些被赋值的元素。对于[稀疏数组](/zh-CN/docs/Web/JavaScript/Guide/Indexed_collections#稀疏数组)来说，空槽行为和 `undefined` 相同。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 15]
```diff
@@ -27,38 +27,36 @@ findLast(callbackFn, thisArg)
...
 - `callbackFn` 不会访问在调用 `findLast()` 开始后才添加到数组中的任何元素。
 - 给已访问过的索引重新赋值将不会被 `callbackFn` 重新访问。
-- 给初始的范围外的索引赋值，其将不会被 `callbackFn` 访问。
-- 如果 `callbackFn` 更改了数组中现有的、尚未访问的元素，则其传递给 `callbackFn` 的值将是 `findLast()` 访问该元素索引时的值。
-- 仍然会访问已{{jsxref("Operators/delete", "删除")}}的元素。
+- 如果 `callbackFn` 更改了数组中现有的、尚未访问的元素，则其传递给 `callbackFn` 的值将是 `findLast()` 访问该元素索引时的值。[已删除](/zh-CN/docs/Web/JavaScript/Reference/Operators/delete)的元素会被当做 `undefined` 来访问。
 
 >**警告：** 上一段描述的并发修改的情况经常导致难以理解的代码，通常应该避免（特殊情况除外）。
 
+`findLast()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它只期望 `this` 值具有 `length` 属性和整数键的属性
```
Comment (fwqaaq) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:58]:
```suggestion
`findLast()` 方法是[通用的](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#通用数组方法)。它只期望 `this` 值具有 `length` 属性和整数键的属性。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md (hunk 13 above)
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:38]: 下面更新了，这行忘了
```suggestion
  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的值。详见[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#迭代方法)
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md [hunk 16]
```diff
@@ -144,22 +149,30 @@ array.findLast(function(value, index) {
...
+// Visited index 4 with value undefined
+// Visited index 3 with value undefined
+// Visited index 2 with value undefined
+// Visited index 1 with value 1
+// Visited index 0 with value 0
+```
+
+### 在非数组对象上调用 findLast()
+
+`findLast()` 方法读取 `this` 的 `length` 属性，然后访问每个整数索引
```
Comment (fwqaaq) [inline files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:164]:
```suggestion
`findLast()` 方法读取 `this` 的 `length` 属性，然后访问每个整数索引。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md (hunk 13 above)
Comment (robert-jx) [reply files/zh-cn/web/javascript/reference/global_objects/array/findlast/index.md:39]:
```suggestion
  - : 执行 `callbackFn` 时，用作 {{jsxref("Operators/this", "this")}} 的值。详见[迭代方法](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#迭代方法)。
```

**Review by yin1999 — APPROVED** [review]
Thanks. Welcome aboard :tada:

**Pull request merged**